src/arkode/arkode_interp.c

```diff
@@ -1036,29 +1026,18 @@ void arkInterpPrintMem_Lagrange(ARKInterp I, FILE* outfile)
   ---------------------------------------------------------------*/
 int arkInterpSetDegree_Lagrange(ARKodeMem ark_mem, ARKInterp I, int degree)
 {
-  /* if this degree is already stored, just return */
-  if (abs(degree) + 1 == LINT_NMAX(I)) { return (ARK_SUCCESS); }
-
-  /* on positive degree, check for allowable value and overwrite stored degree */
-  if (degree >= 0)
+  if (degree > ARK_INTERP_MAX_DEGREE)
   {
-    if (degree > ARK_INTERP_MAX_DEGREE)
-    {
-      arkProcessError(ark_mem, ARK_INTERP_FAIL, __LINE__, __func__, __FILE__,
-                      "Illegal degree specified.");
-      return (ARK_ILL_INPUT);
-    }
-
-    LINT_NMAX(I) = degree + 1;
-    return (ARK_SUCCESS);
+    arkProcessError(ark_mem, ARK_INTERP_FAIL, __LINE__, __func__, __FILE__,
+                    "Illegal degree specified.");
+    return ARK_ILL_INPUT;
   }
 
-  /* on negative degree, check for allowable value and update stored degree */
-  degree = -degree;
-  if (degree > ARK_INTERP_MAX_DEGREE) { degree = ARK_INTERP_MAX_DEGREE; }
-  LINT_NMAX(I) = SUNMIN(LINT_NMAX(I), degree + 1);
+  if (degree < 0) { degree = ARK_INTERP_MAX_DEGREE; }
```

Comment:
Same comment here about the possibility of negative input values, and the comment block preceding the function.
```suggestion
```